cocotb/share/makefiles/Makefile.pylib.Msys

```diff
@@ -54,5 +54,30 @@ empty :=
 PYTHON_VERSION=$(subst $(period),$(empty),$(LOCAL_PYTHON_VERSION))
 PYTHON_INCLUDEDIR := $(shell $(PYTHON_BIN) -c 'import distutils.sysconfig; print( distutils.sysconfig.get_python_inc() )')
 
-PYLIBS = -lPython$(PYTHON_VERSION)
-PYTHON_DYN_LIB = Python$(PYTHON_VERSION).dll
+SYSCONFIG_LDLIBRARY = $(shell $(PYTHON_BIN) -c 'from distutils import sysconfig; print( sysconfig.get_config_var("LDLIBRARY") )')
+
+ifeq ($(SYSCONFIG_LDLIBRARY),None)
+    PYLIBS := -lPython$(PYTHON_VERSION)
+else
+    PYLIBS:=$(shell $(PYTHON_BIN) -c 'from distutils import sysconfig; import os; print("-l"+os.path.splitext(sysconfig.get_config_var("LDLIBRARY"))[0][3:])')
+endif
+
+# Some simulator accept only paths in form of "c:/../.."
+ifneq ($(VERILOG_SOURCES),)
+    VERILOG_SOURCES := $(shell cygpath -m $(VERILOG_SOURCES))
+endif
+ifneq ($(VHDL_SOURCES),)
+    VHDL_SOURCES := $(shell cygpath -m $(VHDL_SOURCES))
+endif
+
+PWD ?= $(shell pwd)
+
+# sys.path in form "/c/../:/c/.."
+PYTHON_SYS_PATH := $(shell $(PYTHON_BIN) -c "import sys, os; print(':'.join(['/'+dir.replace('\\\ ',' ').replace(os.sep,'/').replace(':','') for dir in sys.path]))")
+PYTHONPATH_ESCAPE := $(shell echo "$(PYTHON_SYS_PATH)" | sed 's/(/\\(/g' | sed 's/)/\\)/g' | sed 's/ /\\ /g')
+PYTHONPATH := $(PYTHONPATH_ESCAPE)
```

Comment:
Can't you do this escaping within the python? I don't understand what's going on here

Reply:
Yes. Also prefer python ;-). It was there before like this. Atlully I am not sure if not add some command to `cooctb-config` to return all of this (would make this less ugly)?

Reply:
I tried to do this in line with python but it started to look even more ugly. I have added a comment. If you have some good idea let me know.

Reply:
This whole operation still seems wrong to me, why are we copying the python import path to the system binary path?

Reply:
> It was there before like this

I completely missed this when reviewing this code. It might be worth adding to the commit message that much of this code was just moved

Reply:
I think it should be sufficient to use `pipes.quote` / `shlex.quote`. So:
```define PYTHON_PATH_SOURCE=
import sys
import os
try:
    from shlex import quote
except ImportError:
    from pipes import quote
print(quote(':'.join([
    '/'+dir.replace(os.sep,'/').replace(':','')
    for dir in sys.path
])))
endef

# note single quotes
PYTHONPATH_ESCAPE=$(shell python -c '$(PYTHON_PATH_SOURCE)')

Reply:
Are we sure we want such script to be embedded in the `Makefile`?

Reply:
I think it's fine - the script is very simple, and embedding it on multiple lines is better than wrapping it all onto one line.

Reply:
OK.
What you think about adding `cocotb-config --pythonpath` and `cocotb-config --path` ?

Reply:
This has the advantage that you can have a verbose/debug mode that explains what environment it detected etc.

Reply:
Any conclusion about what would be best:

1. leave as is/was (`sed 's/(/\\(/g' | sed 's/)/\\)/g' | sed 's/ /\\ /g')`)
2. add a few line python script to Makefile
3. add `cocotb-config --pythonpath` and `cocotb-config --path`

@imphil Maybe you could pick.